### PR TITLE
gradience: init at 0.4.1

### PR DESCRIPTION
**Pull request by getchoo**
###### Description of changes

[gradience](https://gradienceteam.github.io/) is a tool for customizing libadwaita applications and the adw-gtk3 theme, with a wide variety of official and community made pallets. this makes theming on desktop environments such as gnome much more streamlined and approachable, with it even supporting the adw-gtk3 theme to add more consistency across your apps.

[material-color-utilities](https://github.com/avanisubbiah/material-color-utilities-python) is a python port of the original [material-color-utilities](https://github.com/material-foundation/material-color-utilities) and a dependency of gradience.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by 0xMRTT** [comment]
Duplicated of https://github.com/NixOS/nixpkgs/pull/189542

File: pkgs/development/python-modules/material-color-utilities/default.nix [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  };
+
+  buildInputs = [ poetry-core ];
+  propagatedBuildInputs = [ pillow regex ];
+
+  meta = with lib; {
+    description = "Python port of material-color-utilities used for Material You colors";
+    license = licenses.asl20;
+    maintainers = with maintainers; [ getchoo ];
+    platforms = platforms.all;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/material-color-utilities/default.nix:26]:
```suggestion
```
buildPythonPackage defaults to linux + darwin

File: pkgs/development/python-modules/material-color-utilities/default.nix [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  pname = "material-color-utilities";
+  version = "0.1.5";
+
+  src = fetchPypi {
+    pname = "${pname}-python";
+    inherit version;
+    sha256 = "sha256-PG8C585wWViFRHve83z3b9NijHyV+iGY2BdMJpyVH64=";
+  };
+
+  buildInputs = [ poetry-core ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/material-color-utilities/default.nix:19]:
```suggestion
  nativeBuildInputs = [ poetry-core ];
```

File: pkgs/development/python-modules/material-color-utilities/default.nix [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  propagatedBuildInputs = [ pillow regex ];
+
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/material-color-utilities/default.nix:21]: Missing pythonImportsCheck

File: pkgs/development/python-modules/material-color-utilities/default.nix [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  meta = with lib; {
+    description = "Python port of material-color-utilities used for Material You colors";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/material-color-utilities/default.nix:23]: Missing homepage

File: pkgs/development/python-modules/material-color-utilities/default.nix [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+, pillow
+, regex
+,
+}:
+buildPythonPackage rec {
+  pname = "material-color-utilities";
+  version = "0.1.5";
+
+  src = fetchPypi {
+    pname = "${pname}-python";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/material-color-utilities/default.nix:14]:
```suggestion
    pname = "material-color-utilities-python";
```

File: pkgs/development/python-modules/material-color-utilities/default.nix [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+    inherit version;
+    sha256 = "sha256-PG8C585wWViFRHve83z3b9NijHyV+iGY2BdMJpyVH64=";
+  };
+
+  buildInputs = [ poetry-core ];
+  propagatedBuildInputs = [ pillow regex ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/material-color-utilities/default.nix:20]:
```suggestion
  propagatedBuildInputs = [ pillow regex ];

  # no tests
  doCheck = false;
```

File: pkgs/applications/misc/gradience/default.nix [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+  src = fetchFromGitHub {
+    owner = "GradienceTeam";
+    repo = "Gradience";
+    rev = version;
+    sha256 = "sha256-xR3wPU0ax9U4995GckC8UGJqrUErd+jS5z3D/jWCdXQ=";
+  };
+
+  nativeBuildInputs =
+    [
+      meson
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/gradience/default.nix:46]:
```suggestion
  nativeBuildInputs = [
    meson
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/applications/misc/gradience/default.nix [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+    anyascii
+    cssutils
+    jinja2
+    lxml
+    material-color-utilities
+    pillow
+    pluggy
+    regex
+    svglib
+    Yapsy
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/gradience/default.nix:29]:
```suggestion
    yapsy
```

**Comment by SuperSandro2000** [comment]
Drafting until #189542 is merged. Please rebase after that if there are some improvements left.

**Pull request closed without merging**